x-pack/elastic-agent/pkg/composable/providers/kubernetes/config.go

```diff
@@ -9,24 +9,53 @@ package kubernetes
 
 import (
 	"time"
+
+	"github.com/elastic/beats/v7/libbeat/logp"
 )
 
 // Config for kubernetes provider
 type Config struct {
+	Scope     string    `config:"scope"`
+	Resources Resources `config:"resources"`
+}
+
+// Resources config section for resources' config blocks
+type Resources struct {
+	Pod     *ResourceConfig `config:"pod"`
+	Node    *ResourceConfig `config:"node"`
+	Service *ResourceConfig `config:"service"`
```

Comment:
Does it make sense to have `kube_config` and `node` for every type? Is the idea here that you can get pods from one cluster and different nodes from another cluster? Or is this needed for scoped permissions in each `kube_config`?

This also will make the configuration of the provider change, which is a breaking configuration change.

Now it will be:

```
providers:
  kubernetes:
    pod:
     node: ${HOSTNAME}
    node:
      node: ${HOSTNAME}
```

Reply:
Yeap, I had in mind to provide such kind of flexibility and give the option for possible different permissions per resource. However the generic use case here is with `inCluster` configs which means that after the improvements of https://github.com/elastic/beats/pull/26947, `kube_config` and `node` settings won't be needed in the general use case since defaults would be enough.

Reply:
Could we still have a top-level defaults for all of these options? So it works as it does today, but if you want to provide specifics for each resource type you can? That would prevent changing what we have today as it would stay working with the same configurations, as well as provide the flexibility that you are adding here.